src/Forum/ForumServiceProvider.php

```diff
@@ -69,21 +82,16 @@ public function register()
                 $app->tagged(Reporter::class)
             ));
 
-            $pipe->pipe($app->make(HttpMiddleware\ParseJsonBody::class));
-            $pipe->pipe($app->make(HttpMiddleware\CollectGarbage::class));
-            $pipe->pipe($app->make(HttpMiddleware\StartSession::class));
-            $pipe->pipe($app->make(HttpMiddleware\RememberFromCookie::class));
-            $pipe->pipe($app->make(HttpMiddleware\AuthenticateWithSession::class));
-            $pipe->pipe($app->make(HttpMiddleware\CheckCsrfToken::class));
-            $pipe->pipe($app->make(HttpMiddleware\SetLocale::class));
-            $pipe->pipe($app->make(HttpMiddleware\ShareErrorsFromSession::class));
+            foreach ($this->app->make('flarum.forum.middleware') as $middleware) {
+                $pipe->pipe($this->app->make($middleware));
+            }
 
             event(new ConfigureMiddleware($pipe, 'forum'));
 
             return $pipe;
         });
 
-        $this->app->afterResolving('flarum.forum.middleware', function (MiddlewarePipe $pipe) {
+        $this->app->afterResolving('flarum.forum.handler', function (MiddlewarePipe $pipe) {
```

Comment:
IMO, we can now get rid of this `afterResolving` call. This was previously necessary to let the extender add middleware, but that now uses a different approach. So, could you move this `$pipe->pipe()` call to right before the `return $pipe` in line 91?